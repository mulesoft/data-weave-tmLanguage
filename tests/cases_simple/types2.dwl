type X = () -> HashMapSetOperation<T>
type X = <T>(key: String, value: T) -> HashMapSetOperation<T>
type HttpServer = {
  running: Boolean,
  port: Number,
  host: String,
  stop: () -> Boolean
}