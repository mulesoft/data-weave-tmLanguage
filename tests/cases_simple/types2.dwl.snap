>type X = () -> HashMapSetOperation<T>
#^^^^ source.data-weave storage.type.dw
#    ^ source.data-weave
#     ^ source.data-weave entity.name.variable.dw
#      ^ source.data-weave
#       ^ source.data-weave keyword.operator.assignment.dw
#        ^^^^ source.data-weave
#            ^^ source.data-weave keyword.operator.case.dw
#              ^ source.data-weave
#               ^^^^^^^^^^^^^^^^^^^ source.data-weave entity.name.variable.dw
#                                  ^ source.data-weave keyword.operator.comparison.dw
#                                   ^ source.data-weave entity.name.variable.dw
#                                    ^ source.data-weave keyword.operator.comparison.dw
>type X = <T>(key: String, value: T) -> HashMapSetOperation<T>
#^^^^ source.data-weave storage.type.dw
#    ^ source.data-weave
#     ^ source.data-weave entity.name.variable.dw
#      ^ source.data-weave
#       ^ source.data-weave keyword.operator.assignment.dw
#        ^ source.data-weave
#         ^ source.data-weave keyword.operator.comparison.dw
#          ^ source.data-weave entity.name.variable.dw
#           ^ source.data-weave keyword.operator.comparison.dw
#            ^ source.data-weave
#             ^^^ source.data-weave entity.name.variable.dw
#                ^ source.data-weave keyword.operator.declaration.dw
#                 ^ source.data-weave
#                  ^^^^^^ source.data-weave entity.name.variable.dw
#                        ^^ source.data-weave
#                          ^^^^^ source.data-weave entity.name.variable.dw
#                               ^ source.data-weave keyword.operator.declaration.dw
#                                ^ source.data-weave
#                                 ^ source.data-weave entity.name.variable.dw
#                                  ^^ source.data-weave
#                                    ^^ source.data-weave keyword.operator.case.dw
#                                      ^ source.data-weave
#                                       ^^^^^^^^^^^^^^^^^^^ source.data-weave entity.name.variable.dw
#                                                          ^ source.data-weave keyword.operator.comparison.dw
#                                                           ^ source.data-weave entity.name.variable.dw
#                                                            ^ source.data-weave keyword.operator.comparison.dw
>type HttpServer = {
#^^^^ source.data-weave storage.type.dw
#    ^ source.data-weave
#     ^^^^^^^^^^ source.data-weave entity.name.variable.dw
#               ^ source.data-weave
#                ^ source.data-weave keyword.operator.assignment.dw
#                 ^^^ source.data-weave
>  running: Boolean,
#^^ source.data-weave
#  ^^^^^^^ source.data-weave entity.name.variable.dw
#         ^ source.data-weave keyword.operator.declaration.dw
#          ^ source.data-weave
#           ^^^^^^^ source.data-weave entity.name.variable.dw
#                  ^^ source.data-weave
>  port: Number,
#^^ source.data-weave
#  ^^^^ source.data-weave entity.name.variable.dw
#      ^ source.data-weave keyword.operator.declaration.dw
#       ^ source.data-weave
#        ^^^^^^ source.data-weave entity.name.variable.dw
#              ^^ source.data-weave
>  host: String,
#^^ source.data-weave
#  ^^^^ source.data-weave entity.name.variable.dw
#      ^ source.data-weave keyword.operator.declaration.dw
#       ^ source.data-weave
#        ^^^^^^ source.data-weave entity.name.variable.dw
#              ^^ source.data-weave
>  stop: () -> Boolean
#^^ source.data-weave
#  ^^^^ source.data-weave entity.name.variable.dw
#      ^ source.data-weave keyword.operator.declaration.dw
#       ^^^^ source.data-weave
#           ^^ source.data-weave keyword.operator.case.dw
#             ^ source.data-weave
#              ^^^^^^^ source.data-weave entity.name.variable.dw
>}
#^^ source.data-weave